web-server/v0.4/src/routes/Dashboard/RunComparison.js

```diff
@@ -38,6 +56,16 @@ export default class RunComparison extends ReactJS.Component {
       graphKeys: [],
       tableData: [],
       loading: false,
+      generatingPdf: false,
+      modal: {
+        visible: false,
+      },
+      checked: ['details', 'summary', 'table', 'timeseries'],
```

Comment:
What do you think about pulling the behaviors of `checked` and `defaultChecked` together? Since we initialize the checked array with the default values for pdf creation, we can reference this array in the export modal for both cases where all values are selected and where the user makes changes to the selection.

Reply:
Can you please update the state names to reflect its intended use case within the code? E.g. `selectedComponents` provides further insight into what is being selected over `checked`.

Reply:
> What do you think about pulling the behaviors of `checked` and `defaultChecked` together? Since we initialize the checked array with the default values for pdf creation, we can reference this array in the export modal for both cases where all values are selected and where the user makes changes to the selection.

Actually what I am trying to do is, when the user selects all the divisions, I am considering the id of the whole page, so that while downloading the PDF, the system don't need to go through loop. And by default also(when no division is selected or all divisions selected), it will download whole page when clicked on save. I am renaming both the  arrays to defaultComponents (All the divisions selected by default) and selectedComponents(selected divisions after change of switch).